sync/src/main/java/tech/pegasys/artemis/sync/SyncManager.java

```diff
@@ -55,17 +62,33 @@ public SyncManager(
     return peerSync
         .sync()
         .thenCompose(
-            result ->
-                result != PeerSyncResult.SUCCESSFUL_SYNC
-                    ? executeSync()
-                    : CompletableFuture.completedFuture(null));
+            result -> {
+              if (result != PeerSyncResult.SUCCESSFUL_SYNC) {
+                return executeSync();
+              } else {
+                syncActive.set(false);
```

Comment:
We shouldn't set this to false here - just leave it to `sync`.

Reply:
done.